### ISSUE TITLE
Maize doesn't behave when SWIM is present

### ISSUE THREAD

**Issue by hol430** [APSIMInitiative/APSIMClassic#1839]
The following issues have been observed in simulations with the 7.10 maize model when SWIM is present:

- Ep is always negative
- Soil moisture (by layer) is generally higher than the old model (same biomass production, weather, soil) 
- Greg told me that he never tested 7.10 with SWIM (told me that back in 2018 June while we were running the APSIM workshop). 
- Missing USA cultivars in xml
- Missing some params in xml (heat stress)
- Source code bug with stem vs rachis dm partitioning near flowering stage

The current workaround is to use the maize model from 7.9 but it would be good if this wasn't necessary.

**Comment by hol430** [comment]
@jbrider I've had a quick look into why EP is negative. It appears that we expect the variable dltSwDep to be negative. When using soilwat, it's calculated like this:

```c++
dltSwDep[layer] = -1 * divide(supply[layer], totalSupply, 0.0) * swDemand;
```

When using swim we just fetch it via the science API:

```c++
scienceAPI.get("uptake_water_maize", "",1, dltSwDep);
```

But it comes in as positive, and we don't multiply it by -1. When calculating other variables like uptake and EP, we always multiply dltSwDep by -1, which means that when we're using swim, EP and uptake end up negative. It seems like the obvious fix here would be to just multiply the incoming dltDwDep by -1 when using swim, so that all other calculations can remain the same:

```c++
scienceAPI.set("dlt_sw_dep", "mm", dltSwDep);
for(unsigned i=0;i < dltSwDep.size();i++)
    dltSwDep[i] *= -1;
```

**Comment by jbrider** [comment]
quick fix - sounds great.
hopefully that fixes the higher soil moisture as well...

**Comment by hol430** [comment]
Looks like there is no code in sorghum or MaizePioneer to even check if swim is present. Do we want them to basically do what maize does?

**Comment by jbrider** [comment]
Yes - hopefully that should be straight forward?

**Comment by hol430** [comment]
Yep, should be. I will create a new issue for sorghum/Maizepioneer so that I can resolve the issue and get an installer out.
